src/routes/fetch-posts.e2e.test.js

```diff
@@ -0,0 +1,187 @@
+				.get('/api/v1/feed/')
+				.expect(401)
+		
+		) 
+
+		it('should return 400 if isFlagged is not a boolean', async () => 
+			fixtures.api()
+				.get(`/api/v1/feed?isFlagged=${'string'}`)
+				.set('Authorization', `Bearer ${accessToken}`)
+				.expect(400)
+		
+		) 
+
+		it('should return 403 if non-admin fetches flagged posts', async () => {
+			const nonAdminUser = await fixtures.insertUser({
+				role : 'user'
+			}) 
+			const nonAdminAccessToken = fixtures.generateAccessToken(
+				{user :  { id: nonAdminUser.id, email: nonAdminUser.email }}
+			)
+			const expectedError = {
+				"status":"failed",
+				"message":"Only admin can create users"
+			}
+			return fixtures.api()
+				.get(`/api/v1/feed?isFlagged=${true}`)
+				.set('Authorization', `Bearer ${nonAdminAccessToken}`)
+				.expect(403 , expectedError)
+		
+		}) 
+
+	})
+	
+	describe('Success', () => {
+
+		const numberOfPosts = 3
+		let insertedPosts
+
+		beforeEach(async () =>{
+
+			await resetDBTable('posts')
+			insertedPosts= await fixtures.insertMultiplePosts(
+				user.id,
+				numberOfPosts
+			)
+			
+		})
+
+		it('should return 200 if posts are fetched', async () =>
+			
+			fixtures.api()
+				.get(`/api/v1/feed`)
+				.set('Authorization', `Bearer ${accessToken}`)
+				.expect(200)
+		)
+
+		it('should return the right number of posts', async() =>
+			
+			fixtures.api()
+				.get(`/api/v1/feed`)
+				.set('Authorization', `Bearer ${accessToken}`)
+				.then(res => {
+
+					expect(res.body.data).to.have.deep.members(
+
+						insertedPosts.map((post) =>({
+							createdOn: post.createdAt.toISOString(),
+							gifId : post.id,
+							imageUrl: post.content,
+							likesCount : post.likesCount,
+							published : post.published,
+							title : post.title,
+							user: {
+								userId: user.id,
+								fullName: `${user.firstName} ${user.lastName}`,
+								profilePictureUrl : user.profilePictureUrl,
+								email : user.email
+							}
+							
+						})))
+				}) 
+		)	
+
+		it('should return the right number of Flagged posts', async() =>{
+			const post = insertedPosts[0]
+			await fixtures.insertPostFlag({
+				userId : user.id ,
+				postId : post.id
+			})
+		
+
+			return fixtures.api()
+				.get(`/api/v1/feed?isFlagged=${true}`)
+				.set('Authorization', `Bearer ${accessToken}`)
+				.then(res => {
+					
+					expect(res.body.data[0].flagsCount).to.eql(1)
+					delete res.body.data[0].flagsCount
+
+					expect(res.body.data[0]).to.eql({
+	
+						createdOn: post.createdAt.toISOString(),
+						gifId : post.id,
+						imageUrl: post.content,
+						likesCount : post.likesCount,
+						published : post.published,
+						title : post.title,
+						user: {
+							userId: user.id,
+							fullName: `${user.firstName} ${user.lastName}`,
+							profilePictureUrl : user.profilePictureUrl,
+							email : user.email
+						}
+						
+							
+					})
+				}) 
+		})
+
+		it('should return the right number of unFlagged posts', async() =>{
```

Comment:
here I don't see us checking the number of unflagged posts like the description says

```suggestion
		it('should return the right number of unflagged posts', async() =>{
```